pkgs/development/libraries/qt-5/5.15/default.nix

```diff
@@ -313,7 +313,9 @@ let
 
       qmake = callPackage ({ qtbase }: makeSetupHook {
         name = "qmake-hook";
-        propagatedBuildInputs = [ qtbase.dev ];
+        ${if stdenv.buildPlatform == stdenv.hostPlatform
+          then "propagatedBuildInputs"
+          else "depsTargetTargetPropagated"} = [ qtbase.dev ];
```

Comment:
Seems wrong, the binaries in qtbase won't be in `PATH`

Reply:
Remember, `qtbase` is both a build tool (`qmake`) and a library (`libQt5Base`).  The line above deals with the library.

Reply:
Are you sure it deals with only the library? https://github.com/nixpkgs/nixpkgs/blob/753bf0e2933c4d66c0eb8481b60157419bae3629/pkgs/development/libraries/qt-5/hooks/qmake-hook.sh#L37

Reply:
> Normally no, except it's QT, and QT has both qmake-which-runs-on-the-buildPlatform as well as libQtBase-which-is-built-for-the-hostPlatform mashed into the same package.

Gotta add comments ..